### PR TITLE
mitmproxy: fix macos build

### PR DESCRIPTION
**Pull request by boltzmannrain**
## Description of changes

https://hydra.nixos.org/build/240114924/nixlog/1
```
error: linking with `/nix/store/sa6hywsm1mqfyd1xakyzv4ljjsb3hawh-clang-wrapper-11.1.0/bin/cc` failed: exit status: 1
  |
...
  = note: ld: library not found for -liconv
          clang-11: error: linker command failed with exit code 1 (use -v to see invocation)
```

after adding libiconv it starts to fail with
```
  File "/nix/store/ncs3h7zqjl3cl2cwflka40rrirb6qg1m-python3.11-mitmproxy-rs-0.3.11/lib/python3.11/site-packages/mitmproxy_rs/__init__.py", line 1, in <module>
    from .mitmproxy_rs import *
ModuleNotFoundError: No module named 'mitmproxy_macos'
```
so it needs extra module
https://github.com/mitmproxy/mitmproxy_rs/tree/main/mitmproxy-macos

So the fixes are
- add `libiconv` dependency to mitmproxy-rs
- add `mitmproxy-macos` python package and add it to dependencies of `mitmproxy-rs` and `mitmproxy` itself (otherwise import fails, looks like the import is sneaky because it is platform-conditional)

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [X] aarch64-linux
  - [X] x86_64-darwin
  - [X] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

ZHF: #265948

### CODE REVIEW

File: pkgs/development/python-modules/mitmproxy-macos/default.nix [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    rev = version;
+    hash = "sha256-V6LUr1jJiTo0+53jipkTyzG5JSw6uHaS6ziyBaFbETw=";
+  };
+
+  sourceRoot = "${src.name}/mitmproxy-macos";
+  pythonImportsCheck = [ "mitmproxy_macos" ];
+  nativeBuildInputs = [
+    hatchling
+  ];
+}
```
Comment (wegank) [inline pkgs/development/python-modules/mitmproxy-macos/default.nix:23]: Would you please add `meta` to the package?

Comment (boltzmannrain) [reply pkgs/development/python-modules/mitmproxy-macos/default.nix:32]: Of course, missed that part while copying from mitmproxy-rs, added now

File: pkgs/development/python-modules/mitmproxy-macos/default.nix [hunk 2]
```diff
@@ -0,0 +1,32 @@
+{  buildPythonPackage
```
Comment (wegank) [inline pkgs/development/python-modules/mitmproxy-macos/default.nix:1]:
```suggestion
{ lib
, buildPythonPackage
```

File: pkgs/development/python-modules/mitmproxy-macos/default.nix [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  nativeBuildInputs = [
+    hatchling
+  ];
+
+  meta = with lib; {
+    description = "The MacOS Rust bits in mitmproxy";
+    homepage = "https://github.com/mitmproxy/mitmproxy_rs/tree/main/mitmproxy-macos";
+    changelog = "https://github.com/mitmproxy/mitmproxy_rs/blob/${src.rev}/CHANGELOG.md";
+    license = licenses.mit;
+    maintainers = with maintainers; [ fab ];
```
Comment (wegank) [inline pkgs/development/python-modules/mitmproxy-macos/default.nix:30]: Would you like to add yourself as a maintainer?

Comment (boltzmannrain) [reply pkgs/development/python-modules/mitmproxy-macos/default.nix:30]: Can do though don't have experience with nixpkgs maintainers process

Comment (wegank) [reply pkgs/development/python-modules/mitmproxy-macos/default.nix:30]: Just follow [the section in CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#commit-conventions). Also, I think fabaff's approval for the maintenance would be necessary.

Comment (boltzmannrain) [reply pkgs/development/python-modules/mitmproxy-macos/default.nix]: Added entry to https://github.com/NixOS/nixpkgs/blob/master/maintainers/maintainer-list.nix and to meta.maintainers

**Review by wegank — APPROVED** [review]

**Review by SuperSandro2000 — APPROVED** [review]
🎉

File: pkgs/development/python-modules/mitmproxy-macos/default.nix (hunk 3 above)
Comment (wegank) [reply pkgs/development/python-modules/mitmproxy-macos/default.nix]: Removed fabaff from the new package.

**Pull request merged**